core/storage/user/gae_models.py

```diff
@@ -97,6 +97,95 @@ def get_by_normalized_username(cls, normalized_username):
             cls.normalized_username == normalized_username).get()
 
 
+class ActivitiesCompletedByLearnerModel(base_models.BaseModel):
+    """Keeps track of all the explorations and collections completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    # IDs of all the explorations completed by the user.
+    completed_exploration_ids = ndb.StringProperty(repeated=True, indexed=True)
+    # IDs of all the collections completed by the user.
+    completed_collection_ids = ndb.StringProperty(repeated=True, indexed=True)
+
+
+class IncompleteCollectionsModel(base_models.BaseModel):
```

Comment:
Is there a reason these are stored separately for explorations and collections, whereas completed activities are stored together? Seems a bit inconsistent to me.